inference-engine/cmake/plugins/plugins.cmake

```diff
@@ -56,11 +56,9 @@ function(ie_add_plugin)
     target_compile_definitions(${IE_PLUGIN_NAME} PRIVATE IMPLEMENT_INFERENCE_ENGINE_PLUGIN)
 
     if(TARGET IE::inference_engine_plugin_api)
-        target_link_libraries(${IE_PLUGIN_NAME} PRIVATE IE::inference_engine_plugin_api
-                                                        IE::inference_engine_legacy)
+        target_link_libraries(${IE_PLUGIN_NAME} PRIVATE IE::inference_engine_plugin_api)
     else()
-        target_link_libraries(${IE_PLUGIN_NAME} PRIVATE inference_engine_plugin_api
-                                                        inference_engine_legacy)
+        target_link_libraries(${IE_PLUGIN_NAME} PRIVATE inference_engine_plugin_api)
```

Comment:
@vladimir-paramuzov after this change, `inference_engine_legacy` will not be passed to linker if plugin does not need it (as we discussed in IM). If plugin requires `inference_engine_legacy`, it links the library explicitly in `plugin_folder/CMakeLists.txt`